packages/grafana-ui/src/components/Chart/Tooltip.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { css } from 'emotion';
+import { Portal } from '../Portal/Portal';
+import { TimeZone, Dimensions } from '@grafana/data';
+import { FlotPosition } from '../Graph/types';
+import { TooltipContainer } from './TooltipContainer';
+
+export type TooltipMode = 'single' | 'multi';
+
+// Describes active dimensions user interacts with
+// It's a key-value pair where:
+// - key is the name of the dimension
+// - value is a tuple addresing which column and row from given dimension is active
+export type ActiveDimensions<T extends Dimensions = any> = { [key in keyof T]: [number, number] | null };
+
+export interface TooltipContentProps<T extends Dimensions = any> {
+  // Each dimension is described by array of fields representing it
+  // I.e. for graph there are two dimensions: x and y axis:
+  // { xAxis: [<array of time fields>], yAxis: [<array of value fields>]}
+  // TODO: type this better, no good idea how yet
+  dimensions: T; // Dimension[]
+  activeDimensions?: ActiveDimensions<T>;
+  timeZone: TimeZone;
+  pos: FlotPosition;
+  mode: TooltipMode;
+}
+
+export interface TooltipProps {
+  /** Element used as tooltips content */
+  content?: React.ReactElement<any>;
+
+  /** Optional component to be used as a tooltip content */
+  tooltipComponent?: React.ComponentType<TooltipContentProps>;
```

Comment:
@davkal by default Graph renders GraphTooltip component, but in Explore you can customise that and provide your own component to handle the tooltip content rendering

Reply:
Not sure Explore needs to customize it (at least initially). Should just be a normal tooltip?